src/rt-app.c

```diff
@@ -300,38 +304,49 @@ static int run_event(event_data_t *event, int dry_run,
 		pthread_mutex_unlock(&(rdata->res.barrier.m_obj));
 		break;
 	case rtapp_sig_and_wait:
-		log_debug("signal and wait %s", rdata->name);
+		log_debug("[%d] sync %s", task_idx, rdata->name);
+		log_debug("[%d] sync - signal %s", task_idx, rdata->name);
 		pthread_cond_signal(&(rdata->res.cond.obj));
+		log_debug("[%d] sync - wait %s", task_idx, rdata->name);
 		pthread_cond_wait(&(rdata->res.cond.obj), &(ddata->res.mtx.obj));
+		log_debug("[%d] sync -complete %s", task_idx, rdata->name);
 		break;
 	case rtapp_broadcast:
 		pthread_cond_broadcast(&(rdata->res.cond.obj));
 		break;
 	case rtapp_sleep:
 		{
 		struct timespec sleep = usec_to_timespec(event->duration);
-		log_debug("sleep %d ", event->duration);
-		nanosleep(&sleep, NULL);
+		log_debug("[%d] sleep %d (%ld.%09ld) start", task_idx, event->duration, sleep.tv_sec, sleep.tv_nsec);
+		while(1) {
```

Comment:
this new while(1) loop changes the behavior of sleep event. It's not about enhancing debug print

Reply:
I agree, this is a change - we should probably review that as a separate patch. The intent is for signals not to interrupt the desired sleep time, but this might not be desirable or necessary.

Reply:
change separated out.

Reply:
I've moved this and a couple of other things into a separate branch on my own fork https://github.com/credp/rt-app/tree/features/cleanups